yamux/src/session.rs

```diff
@@ -287,9 +291,8 @@ where
                     if self.last_send_success.elapsed() > TIMEOUT {
                         return Err(io::ErrorKind::TimedOut.into());
                     }
-                    if self.poll_complete()? {
-                        break;
-                    }
+
+                    return Ok(Async::NotReady);
```

Comment:
Should `poll_complete here`, you can see the comment on `poll_complete` and `send_all`